pennylane_lightning/core/src/gates/Gates.hpp

```diff
@@ -100,6 +100,22 @@ static constexpr auto getHadamard() -> std::vector<ComplexT<T>> {
             INVSQRT2<ComplexT, T>(), -INVSQRT2<ComplexT, T>()};
 }
 
+/**
+ * @brief Create a matrix representation of the SX gate data in row-major
+ * format.
+ *
+ * @tparam ComplexT Complex class.
+ * @tparam T Required precision of gate (`float` or `double`).
+ * @return constexpr std::vector<ComplexT<T>> Return constant expression
+ * of SX data.
+ */
+template <template <typename...> class ComplexT, typename T>
+static constexpr auto getSX() -> std::vector<ComplexT<T>> {
```

Comment:
Should `getSX` have an optional `inverse` parameter, e.g. like `getS` below?

Reply:
If so, it is not caught by any test. Could you modify a test such that it is tested?